www/js/services.js

```diff
@@ -102,6 +102,91 @@ angular.module('emission.services', [])
                "Email cancel reported, seems to be an error on android");
         });
     }
+
+    this.dataCollectionSetConfig = function(config) {
+      return new Promise(function(resolve, reject) {
```

Comment:
I don't think you need these - the data collection interface is already promisified.

https://github.com/e-mission/e-mission-data-collection/blob/master/www/datacollection.js#L39